tensilelite/Tensile/TensileInstructions/Pass.py

```
@@ -116,9 +116,16 @@ def _addRegToGraph(item, assignmentDict, params: list, graph, noOpt):
             _setName2RegNum(p, assignmentDict)
             if p.regType == "acc":
                 continue
-            for i in range(p.regIdx, p.regIdx + p.regNum):
-                if graph[p.regType][i] and graph[p.regType][i][-1] == item:
+            for i in range(p.regIdx, p.regIdx + p.regNum):                 
```

I have seen this issue and it breaks tensile tuning (and we have to go and manually remove the breaking cases). So thanks for adding this to handle tensile tuning breaking. However, since this doesn't solve the underlying root-cause I am wondering if it is worth adding a warning?

---

Yes, there's still something else wrong because it shouldn't be empty or out of index.

---

It should throw an error instead of continue